backend/src/bundles/auth/auth.controller.ts

```diff
@@ -135,6 +141,63 @@ class AuthController extends ControllerBase {
             payload: await this.authService.signIn(options.body),
         };
     }
+
+    /**
+     * @swagger
+     * /auth/current-user:
+     *   get:
+     *     tags:
+     *       - Auth
+     *     description: Get the current user based on the provided token
+     *     security:
+     *       - BearerAuth: []
+     *     responses:
+     *       200:
+     *         description: Successful operation
+     *         content:
+     *           application/json:
+     *             schema:
+     *               $ref: '#/components/schemas/User'
+     *       401:
+     *         description: Unauthorized
+     *         content:
+     *           application/json:
+     *             schema:
+     *               $ref: '#/components/schemas/Error'
+     *       404:
+     *         description: User not found
+     *         content:
+     *           application/json:
+     *             schema:
+     *               $ref: '#/components/schemas/Error'
+     */
+
+    private async getCurrentUser(
+        options: ApiHandlerOptions,
+    ): Promise<ApiHandlerResponse> {
+        const [, token] = options.headers.authorization?.split(' ') ?? [];
+
+        if (!token) {
+            throw new HttpError({
+                status: HttpCode.UNAUTHORIZED,
+                message: 'Authorization token is missing',
+            });
+        }
+
+        const user = await this.authService.findByToken(token);
+
+        if (!user) {
+            throw new HttpError({
+                status: HttpCode.NOT_FOUND,
+                message: 'No user found for provided token',
+            });
+        }
```

Comment:
looks like all this logic should be in the service layer

Reply:
I think only this part should be moved 
if (!user) {
            throw new HttpError({
                status: HttpCode.NOT_FOUND,
                message: 'No user found for the provided token',
            });
        }
        
        if (!token)   - is quite logically to leave here

Reply:
you can handle it all in the service. It will be much cleaner. Controller will return data and Service will be handle all magic